### PR TITLE
DT-1232: ⚡ Add in jdbc storage of http sessions

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/config/AuthDbConfig.kt [hunk 1]
```diff
@@ -27,6 +28,7 @@ class AuthDbConfig {
   lateinit var datasource: Map<String, String>
 
   @Bean(name = ["authDataSource"])
+  @SpringSessionDataSource
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/config/AuthDbConfig.kt:31]: annotation to get the correct data source automatically injected

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/config/AuthJdbcHttpSessionConfiguration.kt [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package uk.gov.justice.digital.hmpps.oauth2server.config
+
+import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.context.annotation.Configuration
+import org.springframework.session.jdbc.config.annotation.web.http.JdbcHttpSessionConfiguration
+import org.springframework.transaction.PlatformTransactionManager
+
+@Configuration
+class AuthJdbcHttpSessionConfiguration(
+  @Qualifier("authTransactionManager") transactionManager: PlatformTransactionManager,
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/config/AuthJdbcHttpSessionConfiguration.kt:10]: unfortunately can't do the same with the transaction manager so have to override and configure

File: src/main/resources/db/auth/V5_25__jdbc_http_session.sql [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+);
+
+CREATE UNIQUE INDEX SPRING_SESSION_IX1 ON SPRING_SESSION (SESSION_ID);
+CREATE INDEX SPRING_SESSION_IX2 ON SPRING_SESSION (EXPIRY_TIME);
+CREATE INDEX SPRING_SESSION_IX3 ON SPRING_SESSION (PRINCIPAL_NAME);
+
+CREATE TABLE SPRING_SESSION_ATTRIBUTES (
+                                           SESSION_PRIMARY_ID CHAR(36) NOT NULL,
+                                           ATTRIBUTE_NAME VARCHAR(200) NOT NULL,
+                                           ATTRIBUTE_BYTES IMAGE NOT NULL,
```
Comment (petergphillips) [inline src/main/resources/db/auth/V5_25__jdbc_http_session.sql:19]: this is the schema-sqlserver.sql version copied from the jar file, seems to work okay in h2 as well so didn't bother with separate versions

**Review by SimonMitchellMOJ — APPROVED** [review]
LGTM

**Pull request merged**